src/Themes.vala

```diff
@@ -63,37 +79,30 @@ public class Terminal.Themes {
                 };
                 break;
             case (DARK):
-                string_palette = DARK_PALETTE;
+                string_palette = {
+                    "#073642", "#dc322f", "#859900", "#b58900", "#268bd2", "#ec0048", "#2aa198", "#94a3a5",
+                    "#586e75", "#cb4b16", "#859900", "#b58900", "#268bd2", "#d33682", "#2aa198", "#6c71c4",
+                    "rgba(46, 46, 46, 0.95)", "#a5a5a5", "#839496"
+                };
                 break;
-            default:
+            case (CUSTOM):
                 string_palette = Application.settings.get_string ("palette").split (":");
                 string_palette += Application.settings.get_string ("background");
                 string_palette += Application.settings.get_string ("foreground");
                 string_palette += Application.settings.get_string ("cursor-color");
                 break;
         }
 
-        bool settings_valid = string_palette.length == PALETTE_SIZE;
-
-        var rgba_palette = new Gdk.RGBA[PALETTE_SIZE];
-        for (int i = 0; i < PALETTE_SIZE; i++) {
-            var new_color = Gdk.RGBA ();
-            // Replace invalid color with corresponding one from default palette
-            if (!new_color.parse (string_palette[i])) {
-                critical ("Color %i '%s' is not valid - replacing with default", i, string_palette[i]);
-                string_palette[i] = DARK_PALETTE[i];
-                new_color.parse (DARK_PALETTE[i]);
-                settings_valid = false;
-            }
-
-            rgba_palette[i] = new_color;
-        }
-
-        if (!settings_valid) {
-            /* Remove invalid colors from setting */
-            Application.settings.set_string ("palette", string.joinv (":", string_palette));
-        }
```

Comment:
I don't think this was meant to be removed. This overwrites the setting with the sanitized palette if we detect that any of the values are invalid

Reply:
Reinstated